packages/preview-middleware-client/test/unit/adp/controllers/AddFragment.controller.test.ts

```diff
@@ -98,21 +110,35 @@ describe('AddFragment', () => {
 
             const event = {
                 getSource: jest.fn().mockReturnValue({
-                    getSelectedItem: jest.fn().mockReturnValue({ getText: jest.fn().mockReturnValue('some-text') }),
+                    getSelectedItem: jest.fn().mockReturnValue({ getText: jest.fn().mockReturnValue('someText') }),
                     getSelectedKey: jest.fn().mockReturnValue('0')
                 })
             };
 
             ControlUtils.getControlAggregationByName = jest.fn().mockReturnValue({ 0: {} });
 
+            OverlayRegistry.getOverlay = jest.fn().mockReturnValue({
+                getDesignTimeMetadata: jest.fn().mockReturnValue({
+                    getData: jest.fn().mockReturnValue({
+                        aggregations: { someText:{ specialIndexHandling: 'true' } }
+                    })
+                })
+            });
+
+            addFragment['runtimeControl'] = {
+                getMetadata: jest.fn().mockReturnValue({
+                    getName: jest.fn().mockReturnValue('Toolbar')
+                })
+            } as unknown as ManagedObject;
+
             const setPropertySpy = jest.fn();
             addFragment.model = {
                 setProperty: setPropertySpy
             } as unknown as JSONModel;
 
             addFragment.onAggregationChanged(event as unknown as Event);
 
-            expect(setPropertySpy).toHaveBeenCalledTimes(4);
+            expect(setPropertySpy).toHaveBeenCalledTimes(7);
```

Comment:
This only tests if the function was called, but not if it was called with the correct parameters. Please add checks for the parameters.